playground/DatabaseMigration/DatabaseMigration.MigrationService/ApiDbInitializer.cs

```diff
@@ -0,0 +1,35 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+
+namespace DatabaseMigration.MigrationService;
+
+public class ApiDbInitializer(
+    IServiceProvider serviceProvider,
+    IHostApplicationLifetime hostApplicationLifetime) : BackgroundService
+{
+    public const string ActivitySourceName = "Migrations";
+    private static readonly ActivitySource s_activitySource = new(ActivitySourceName);
+
+    protected override async Task ExecuteAsync(CancellationToken cancellationToken)
+    {
+        using var scope = serviceProvider.CreateScope();
+        var dbContext = scope.ServiceProvider.GetRequiredService<MyDb1Context>();
+
+        await InitializeDatabaseAsync(dbContext, cancellationToken);
+
+        hostApplicationLifetime.StopApplication();
+    }
+
+    private static async Task InitializeDatabaseAsync(MyDb1Context dbContext, CancellationToken cancellationToken)
+    {
+        var strategy = dbContext.Database.CreateExecutionStrategy();
+
+        using var activity = s_activitySource.StartActivity("Migrating database", ActivityKind.Client);
+
+        await strategy.ExecuteAsync(() => dbContext.Database.MigrateAsync(cancellationToken));
```

Comment:
If you are adding retries you also need to wrap `MigrateAsync` in `Database.BeginTransactionAsync()`/`CommitAsync`, otherwise if the connection is interrupted midway it will leave the database in a bad state.